octodns/provider/azuredns.py

```diff
@@ -259,8 +259,21 @@ def _parse_azure_type(string):
     return string.split('/')[-1]
 
 
-def _traffic_manager_suffix(record):
-    return record.fqdn[:-1].replace('.', '-')
+def _root_traffic_manager_name(record):
+    # ATM names can only have letters, numbers and hyphens
+    # replace dots with double hyphens to ensure unique mapping,
+    # hoping that real life FQDNs won't have double hyphens
+    return record.fqdn[:-1].replace('.', '--')
```

Comment:
Naming character restrictions like this are so annoying. Computers don't care what characters are in there... :grin:

Reply:
There go my plans of using emojis for traffic manager names 👎